lang/java/reef-common/src/main/java/org/apache/reef/driver/evaluator/EvaluatorRequest.java

```diff
@@ -55,14 +56,24 @@
                    final List<String> nodeNames,
                    final List<String> rackNames,
                    final String runtimeName) {
+    this(number, megaBytes, cores, nodeNames, rackNames, runtimeName, null);
+  }
+
+  EvaluatorRequest(final int number,
+                   final int megaBytes,
+                   final int cores,
+                   final List<String> nodeNames,
+                   final List<String> rackNames,
+                   final String runtimeName,
+                   final String nodeLabelExpression) {
```

Comment:
They should. And we can probably create a unit test to verify that - at least locally and on YARN